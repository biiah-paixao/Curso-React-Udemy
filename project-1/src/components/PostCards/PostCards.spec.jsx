import { render, screen } from '@testing-library/react';
import { PostCard } from '.';

const mock = {
  title: 'Title',
  body: 'body 1',
  id: 1,
  cover: 'img/img.png',
};

describe('<PostCard />', () => {
  it('should rende the Post correctly', () => {
    render(<PostCard {...mock} />);

    expect(screen.getByRole('img', { name: mock.title })).toHaveAttribute('src', mock.cover);
    expect(screen.getByRole('heading', { name: mock.title })).toBeInTheDocument();
    expect(screen.getByText(mock.body)).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...mock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
