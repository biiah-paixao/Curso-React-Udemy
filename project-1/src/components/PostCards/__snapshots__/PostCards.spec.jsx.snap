// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Title"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      Title
    </h1>
    <p>
      body 1
    </p>
  </div>
</div>
`;
